app/services/report_importer.rb

```diff
@@ -2,14 +2,22 @@ class ReportImporter
   delegate :logger, :to => :Rails
   attr_reader :report
 
+  def self.authorized_smart_proxy_features
+    # When writing your own Report importer, provide feature(s) of authorized Smart Proxies
+    Rails.logger.debug("Importer #{self} does not implement authorized_smart_proxy_features.")
+    []
+  end
+
   def self.import(raw, proxy_id = nil)
     importer = new(raw, proxy_id)
     importer.import
     importer.report
   end
 
-  def self.report_features
-    ['Puppet', 'Chef Proxy']
+  # to be defined in children
+  def report_name_class
+    Foreman::Deprecation.deprecation_warning('1.12', "ReportImport is a supperclass, please use child classes to define 'report_class_name'")
```

Comment:
Is a deprecation warning required here?  I'm trying to see how it might be called, and I think it's only by defining a new STI type anyway, which means you're already using this new feature.